File: Practica final/Napakalaki/src/napakalaki/Monster.java
package napakalaki;

public class Monster {
    private String name;
    private int combatlevel;
    private Prize prize;
    private BadConsequence badConsequence;
    
    public Monster(String name, int combatlevel) {
        this.name = name;
        this.combatlevel = combatlevel;
    }

    public Monster(String name, int level, BadConsequence bc, Prize prize) {
        this.name = name;
        this.combatlevel = level;
        this.prize = prize;
        this.badConsequence = bc;
    }
    
    public String getName(){
        return name;
    }
    public int getCombatLevel(){
        return combatlevel;
    }
    
    public Prize getPrize(){
        return prize;
    }
    
    public BadConsequence getBadConsequence(){
        return badConsequence;
    }
    
    public int getLevelsGained()
    {
        return getPrize().getLevel();
    }
    
    public int getTreasuresGained()
    {
            return getPrize().getTreasures();
    }
    
    public String toString(){
        return "Monster Name = " + name 
                + " CombatLevel = " + Integer.toString(combatlevel) 
                + " Prize = [[ " + prize + " ]]"
                + " badConsequence = [[ " + badConsequence + " ]]";
    }
    
    public boolean nivelSuperiorA(int lvl)
    {
        return ( this.combatlevel > lvl );
    }
    
    
    public boolean MalRolloPerdidaNiveles(){
        return ( badConsequence.getLevels() > 0 
                && badConsequence.getnHiddenTreasures() == 0
                && badConsequence.getnVisibleTreasures() == 0
                && badConsequence.getSHiddenTreasures().isEmpty()
                && badConsequence.getSVisibleTreasures().isEmpty());
    }
    
    public boolean BuenRolloGananciaNivelesSupA(int lvl){
        return ( prize.getLevel() > lvl );
    }
    
    public boolean PerdidaTesoroVisibleOculto(){ //FIXME
        return ( badConsequence.getnHiddenTreasures() != 0 
                || badConsequence.getnVisibleTreasures() != 0 
                    || !badConsequence.getSHiddenTreasures().isEmpty()//Si no tiene ningun elemento en el array de tesoros ocultos
                        || !badConsequence.getSVisibleTreasures().isEmpty());
    }
}

File: Practica final/NapakalakiRuby/lib/monster.rb
# encoding: UTF-8

class Monster
  
  #Atributos
  attr_reader :name, :combat_level, :prize, :badConsequence
  
  #Constructor
  def initialize(name, level, bc, prize)
    @name = name
    @combat_level = level
    @prize = prize
    @badConsequence = bc
  end

  def to_s
    "Monster = #{@name} | CombatLevel = #{@combat_level}" +
      " Prize = [[ #{@prize.to_s} ]]" +
      " badConsequence = [[ #{@badConsequence.to_s} ]]"
  end
  
  def nivelSuperiorA(lvl)
    ( @combat_level > lvl )
  end

  def MalRolloPerdidaNiveles
    ( @badConsequence.levels > 0 && @badConsequence.nHiddenTreasures == 0 && @badConsequence.nVisibleTreasures == 0 && @badConsequence.specificHiddenTreasures.empty? && @badConsequence.specificVisibleTreasures.empty?  )
  end

  def BuenRolloGananciaNivelesSupA(lvl)
    ( @prize.level > lvl )
  end
  
  def PerdidaTesoroVisibleOculto
    ( @badConsequence.nHiddenTreasures != 0 || @badConsequence.nVisibleTreasures != 0 || !@badConsequence.specificHiddenTreasures.empty? || !@badConsequence.specificVisibleTreasures.empty? )
  end
  
end

